src/main/java/com/synopsys/integration/jenkins/scan/bridge/BridgeDownloaderAndExecutor.java

```diff
@@ -0,0 +1,99 @@
+package com.synopsys.integration.jenkins.scan.bridge;
+
+import com.synopsys.integration.jenkins.scan.global.ApplicationConstants;
+
+import hudson.EnvVars;
+import hudson.FilePath;
+import hudson.Platform;
+import hudson.model.TaskListener;
+
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Objects;
+
+public class BridgeDownloaderAndExecutor {
+
+    private final String bridgeZipFileName = "bridge.zip";
+    private final TaskListener listener;
+    private final EnvVars envVars;
+
+    public BridgeDownloaderAndExecutor(TaskListener listener, EnvVars envVars) {
+        this.listener = listener;
+        this.envVars = envVars;
+    }
+
+    public FilePath downloadSynopsysBridge(FilePath downloadFilePath, String bridgeVersion, String bridgeDownloadUrl) {
+        FilePath bridgeZipPath = downloadFilePath.child(bridgeZipFileName);
+        String bridgeUrl;
+        
+        if (isValidVersion(bridgeVersion)) {
+            bridgeUrl = ApplicationConstants.BRIDGE_ARTIFACTORY_URL
+                .concat(bridgeVersion).concat("/")
+                .concat(ApplicationConstants.getSynopsysBridgeZipFileName(getPlatform()));
+        } else if (isValidBridgeDownloadUrl(bridgeDownloadUrl)){
+            bridgeUrl = bridgeDownloadUrl;
+        } else {
+            bridgeUrl = ApplicationConstants.BRIDGE_ARTIFACTORY_URL
+                .concat(ApplicationConstants.SYNOPSYS_BRIDGE_LATEST_VERSION).concat("/")
+                .concat(ApplicationConstants.getSynopsysBridgeZipFileName(ApplicationConstants.PLATFORM_LINUX));
+        }
+
+        if (checkIfBridgeUrlExists(bridgeUrl)) {
+            try {
+                listener.getLogger().println("Downloading synopsys bridge from: ".concat(bridgeUrl));
+                bridgeZipPath.copyFrom(new URL(bridgeUrl));
+                listener.getLogger().println("Synopsys bridge downloaded successfully to: ".concat(bridgeZipPath.getRemote()));
+            } catch (Exception e) {
+                listener.getLogger().println("Synopsys bridge download failed");
+                e.printStackTrace();
+            }
+        } else {
+            listener.getLogger().println("Invalid synopsys bridge download url: " + bridgeUrl);
+        }
+        return bridgeZipPath;
+    }
+
+    public void unzipSynopsysBridge(FilePath bridgeZipPath, FilePath bridgeUnzipPath) {
+        try {
+            listener.getLogger().println("Synopsys bridge zip path: ".concat(bridgeZipPath.getRemote()));
+            bridgeZipPath.unzip(bridgeUnzipPath);
```

Comment:
We may define a specific accessible directory (For default location) and unzip the executables inside that.  We should also have facility to accept a parameter for specifying location.